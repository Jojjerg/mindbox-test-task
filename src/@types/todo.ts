export interface ITodo {
  userId: number;
  id: number;
  title: string;
  completed: boolean;
}

export type TodoContextType = {
  todos: ITodo[];
  filter: IFilter;
  getTodos: () => void;
  addTodo: (title: string) => void;
  updateTodo: (id: number) => void;
  deleteCompletedTodos: () => void;
  changeFilter: (filter: IFilter) => void;
};

export type IFilter = "All" | "Completed" | "Active";
