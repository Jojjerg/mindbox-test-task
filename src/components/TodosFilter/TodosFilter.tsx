import type { IFilter, ITodo } from "../../@types/todo";

import DeleteCompletedButton from "../DeleteCompletedTodosButton/DeleteCompletedTodosButton";
import Filters from "../Filters/Filters";
import React from "react";
import RemainingTodos from "../RemainingTodos/RemainingTodos";

type TodosFilterProps = {
  todos: ITodo[];
  changeFilter: (filter: IFilter) => void;
  deleteCompletedTodos: () => void;
}

const TodosFilter: React.FC<TodosFilterProps> = ({ todos, changeFilter, deleteCompletedTodos }) => {
  return (
    <div className="flex items-center justify-between px-4 py-3 w-full bg-[#282828] border border-transparent border-t-zinc-600 rounded-b-md">
      <RemainingTodos todos={todos}/>
      <Filters changeFilter={changeFilter}/>
      <DeleteCompletedButton deleteCompletedTodos={deleteCompletedTodos} />
    </div>
  );
};

export default TodosFilter;
