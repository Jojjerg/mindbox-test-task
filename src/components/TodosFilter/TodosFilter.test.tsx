import { ITodo, TodoContextType } from "../../@types/todo";
import { render, screen } from "@testing-library/react";

import React from "react";
import TodosFilter from "./TodosFilter";
import { createRenderer } from "react-test-renderer/shallow";

const renderer = createRenderer();

const todos = [
  {
    userId: 1,
    id: 1,
    title: "delectus aut autem",
    completed: false,
  },
  {
    userId: 1,
    id: 2,
    title: "quis ut nam facilis et officia qui",
    completed: false,
  },
  {
    userId: 1,
    id: 3,
    title: "fugiat veniam minus",
    completed: false,
  },
  {
    userId: 1,
    id: 4,
    title: "et porro tempora",
    completed: true,
  },
  {
    userId: 1,
    id: 5,
    title: "laboriosam mollitia et enim quasi adipisci quia provident illum",
    completed: false,
  },
  {
    userId: 1,
    id: 6,
    title: "qui ullam ratione quibusdam voluptatem quia omnis",
    completed: false,
  },
  {
    userId: 1,
    id: 7,
    title: "illo expedita consequatur quia in",
    completed: false,
  },
  {
    userId: 1,
    id: 8,
    title: "quo adipisci enim quam ut ab",
    completed: true,
  },
];

let filter = "All";

const deleteCompletedTodos = () => {
  return todos.filter((todo: ITodo) => !todo.completed);
};

const changeFilter = (newFilter: TodoContextType["filter"]) => {
  filter = newFilter;
  return filter;
};

const defaultComponent = (
  <TodosFilter
    todos={todos}
    changeFilter={changeFilter}
    deleteCompletedTodos={deleteCompletedTodos}
  />
);

describe("<TodosFilter />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
