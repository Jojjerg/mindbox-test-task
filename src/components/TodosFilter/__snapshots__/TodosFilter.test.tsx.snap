// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodosFilter /> should render and match the snapshot 1`] = `
<div
  className="flex items-center justify-between px-4 py-3 w-full bg-[#282828] border border-transparent border-t-zinc-600 rounded-b-md"
>
  <RemainingTodos />
  <Filters />
  <DeleteCompletedTodosButton />
</div>
`;
